iidm/iidm-impl/src/main/java/com/powsybl/iidm/network/impl/AbstractVoltageLevel.java

```diff
@@ -340,6 +342,28 @@ protected String getTypeDescription() {
         return "Voltage level";
     }
 
+    /**
+     * Throw a {@link com.powsybl.commons.PowsyblException} if this voltage level contains at least one {@link Branch} or
+     * one {@link ThreeWindingsTransformer} or one {@link HvdcConverterStation}.
+     */
+    protected void checkRemovability() {
+        Set<ConnectableType> types = EnumSet.of(
+                ConnectableType.LINE,
+                ConnectableType.TWO_WINDINGS_TRANSFORMER,
+                ConnectableType.THREE_WINDINGS_TRANSFORMER);
+
+        for (Connectable connectable : getConnectables()) {
+            ConnectableType type = connectable.getType();
+            if (types.contains(type)) {
+                // Reject lines, 2WT and 3WT
+                throw new AssertionError("The voltage level '" + getId() + "' cannot be removed because of a remaining " + type);
```

Comment:
My first idea was to use this method only in the network reduction. So if this exception is thrown, there is an issue because the algorithm isolate the substation and the voltage level before removing them.
But you’re right, the remove method could be use in many use cases, I will do a fix